homeassistant/components/comfoconnect/sensor.py

```diff
@@ -246,90 +274,65 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     sensors = []
     for resource in config[CONF_RESOURCES]:
-        sensors.append(
-            ComfoConnectSensor(
-                name=f"{ccb.name} {SENSOR_TYPES[resource][ATTR_LABEL]}",
-                ccb=ccb,
-                sensor_type=resource,
-            )
-        )
+        sensors.append(ComfoConnectSensor(ccb=ccb, description=SENSOR_TYPES[resource]))
 
     add_entities(sensors, True)
```

Comment:
This whole section would then need to be rewritten. Probably something like this:
```py
resources = config[CONF_RESOURCES]
entities = [
    ComfoConnectSensor(ccb=ccb, description=description)
    for description in SENSOR_TYPES
    if description.key in resources
]
add_entities(sensors, True)
```